File: php/testcode/benchmark_test_00605.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest00605(BenchmarkRequest $req): BenchmarkResponse {
    $input = $req->param('token');
    if (!preg_match('/^[a-zA-Z0-9_-]+$/', $input)) {
        return BenchmarkResponse::badRequest('Invalid token format');
    }
    header("X-Auth-Token: " . $input);
    return BenchmarkResponse::ok('Authenticated');
}

File: rust/testcode/benchmark_test_00605.rs
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let mut username = req.param("username");
    username = "redacted".to_string();
    log_info(&format!("Login attempt: user={}", username));
    super::shared::BenchmarkResponse::ok("Logged")
}

fn log_info(msg: &str) {
    eprintln!("[INFO] {}", msg);
}
